### ISSUE TITLE
Facade\IgnitionContracts\SolutionProviderRepository not instantiable

### ISSUE THREAD

**Issue by ACvijic** [mateusjunges/laravel-acl#143]
Due to the recent `2.2.0` [update](https://github.com/mateusjunges/laravel-acl/pull/141), which added ignition solutions, I can't get your package to work :sweat_smile: 
At least I think this is the culprit. Here's the stack trace:

```
Generating optimized autoload files
> Illuminate\Foundation\ComposerScripts::postAutoloadDump
> @php artisan package:discover --ansi

   Illuminate\Contracts\Container\BindingResolutionException  : Target [Facade\IgnitionContracts\SolutionProviderRepository] is not instantiable.

  at /opt/vendor/laravel/framework/src/Illuminate/Container/Container.php:979
    975|         } else {
    976|             $message = "Target [$concrete] is not instantiable.";
    977|         }
    978| 
  > 979|         throw new BindingResolutionException($message);
    980|     }
    981| 
    982|     /**
    983|      * Throw an exception for an unresolvable primitive.

  Exception trace:

  1   Illuminate\Container\Container::notInstantiable("Facade\IgnitionContracts\SolutionProviderRepository")
      /opt/vendor/laravel/framework/src/Illuminate/Container/Container.php:813

  2   Illuminate\Container\Container::build("Facade\IgnitionContracts\SolutionProviderRepository")
      /opt/vendor/laravel/framework/src/Illuminate/Container/Container.php:682

  Please use the argument -v to see more details.
Script @php artisan package:discover --ansi handling the post-autoload-dump event returned with error code 1
```

When I revert to version `2.1.1` the bug goes away.

**Comment by mateusjunges** [comment]
@ACvijic I'll take a look at this as soon as possible. If it is not a problem for you, use v2.1.1 for now. Thanks for your report

**Comment by ACvijic** [comment]
@mateusjunges no problem!
Thanks for the quick reply! :slightly_smiling_face:

**Comment by mateusjunges** [comment]
@ACvijic which laravel and php version are you using? I can't reproduce this error.

**Comment by ACvijic** [comment]
@mateusjunges 

Laravel `6.3.0`
PHP `7.2.1`

**Comment by mateusjunges** [comment]
This is myenvironment config, and i still not able to reproduce this error.

**Environment information**
**Laravel version:** 6.3.0
**Laravel locale:** en
**PHP version:** 7.2.19-0ubuntu0.18.04.2

**Package installation**
 - `composer require mateusjunges/laravel-acl`
- `php artisan acl:install`

Are you doing something different?

**Comment by ACvijic** [comment]
I just added it to my `composer.json`, then ran `composer install` - this is the only difference.

I am running my Laravel app in a Docker container, maybe it's something on my end... I'll try to get to the bottom of it next week and I'll keep you posted if I find anything.

**Comment by mateusjunges** [comment]
I'll make tests with  `Docker` tonight. Let me know if you figure out anything new.

**Comment by ReeceM** [comment]
Hi @ACvijic, quick question, in your composer file, have you got `"facade/ignition": "^1.4"` as a composer dev dependency?

**Comment by mateusjunges** [comment]
@ACvijic i will release 2.2.1 to fix this issue in ~10 minutes. Let me know if it solve your problem.

**Comment by ACvijic** [comment]
Sorry to report, #144 didn't fix it for me.

I previously had `mateusjunges/laravel-acl` at `2.1.1` to avoid the bug at version `2.2.0`; To test out @ReeceM 's fix I set `mateusjunges/laravel-acl` to `^2.1`, which brought it to version `2.2.1`, and I didn't change anything else about my `composer.json`. Still the error persisted. Here's the stack trace:

```
localuser@07a60472bf5b:/opt$ composer install
Loading composer repositories with package information
Updating dependencies (including require-dev)
^[[APackage operations: 1 install, 3 updates, 0 removals
  - Installing facade/ignition-contracts (1.0.0): Downloading (100%)         
  - Updating opis/closure (3.4.0 => 3.4.1): Downloading (100%)         
  - Updating nesbot/carbon (2.25.2 => 2.25.3): Downloading (100%)         
  - Updating mateusjunges/laravel-acl (2.1.1 => 2.2.1): Downloading (100%)         
Writing lock file
Generating optimized autoload files
> Illuminate\Foundation\ComposerScripts::postAutoloadDump
> @php artisan package:discover --ansi

   Illuminate\Contracts\Container\BindingResolutionException  : Target [Facade\IgnitionContracts\SolutionProviderRepository] is not instantiable.

  at /opt/vendor/laravel/framework/src/Illuminate/Container/Container.php:979
    975|         } else {
    976|             $message = "Target [$concrete] is not instantiable.";
    977|         }
    978| 
  > 979|         throw new BindingResolutionException($message);
    980|     }
    981| 
    982|     /**
    983|      * Throw an exception for an unresolvable primitive.

  Exception trace:

  1   Illuminate\Container\Container::notInstantiable("Facade\IgnitionContracts\SolutionProviderRepository")
      /opt/vendor/laravel/framework/src/Illuminate/Container/Container.php:813

  2   Illuminate\Container\Container::build("Facade\IgnitionContracts\SolutionProviderRepository")
      /opt/vendor/laravel/framework/src/Illuminate/Container/Container.php:682

  Please use the argument -v to see more details.
Script @php artisan package:discover --ansi handling the post-autoload-dump event returned with error code 1
```

Then I followed @ReeceM 's previous advice:

> Hi @ACvijic, quick question, in your composer file, have you got `"facade/ignition": "^1.4"` as a composer dev dependency?

and added `"facade/ignition": "^1.4"` to the `require-dev` section of my `composer.json`. This thankfully fixed the issue! :tada: 

My guess:

Maybe there is a problem w/ the version of `facade/ignition`... but when I look back on the stack trace that I sent above I see that it installed `facade/ignition-contracts` at version `1.0.0`, and not `facade/ignition`. I think it should be installing `facade/ignition` as it was added in [this commit](https://github.com/mateusjunges/laravel-acl/pull/144/commits/b7cd9b20fbfed9ffe63aaf35f9950ce3d941811d) from #144 , but I don't know why it's not :confused:  .

And, just to recap, for me the fix was adding `"facade/ignition": "^1.4"` to `require-dev`.

Sorry for the long comment. Maybe this is just a bug specific to my configuration. I'm content w/ the fix, and glad that I can use @mateusjunges 's great plugin at its latest & greatest version! :smiley: 
Thanks for looking into this, @ReeceM ! :slightly_smiling_face: I do think that `facade/ignition` is a great addition!

**Comment by ReeceM** [comment]
Hi @ACvijic, glad that you have it working now 😃 .

With regards:
> Maybe there is a problem w/ the version of facade/ignition... but when I look back on the stack trace that I sent above I see that it installed facade/ignition-contracts at version 1.0.0, and not facade/ignition. I think it should be installing facade/ignition as it was added in this commit from #144 , but I don't know why it's not 😕 .

I had my local copy of `laravel-acl` installing `facade/ignition` when I required it into a local app, but this was using a local repository reference in the app's `composer.json` file, but I think that worked only 'cause it pulled in the `facade/ignition` package even thought it's in `require-dev` of laravel-acl. 

But when it pulls from the actual packagist site, composer doesn't pull the dev requirements of a package with it as well, or we would see a lot more of `orchestra/testbench` in everyones vendor file, as that is under `require-dev` in most packages.

_I think then composer didn't pull in the full ignition package as I mistakenly put it under `require-dev` in Mateus' package._

This may need to be changed in case of this package being required into Laravel instances which haven't got the facade/ignition `package` yet.

Glad you got it running though and thank you for the feedback, even if it took a while, we are all busy 👍.
